lms/services/user.py

```diff
@@ -1,24 +1,24 @@
-from dataclasses import dataclass
+from functools import lru_cache
 from typing import Optional
 
-from sqlalchemy.orm import Session
+from lms.models import LTIUser, User
 
-from lms.models import LTIUser
-from lms.models.user import User
-from lms.services.application_instance import ApplicationInstanceService
+
+class UserNotFound(Exception):
+    """The requested User wasn't found in the database."""
 
 
-@dataclass
```

Comment:
Yep. Dataclasses are great but lets not abuse or overuse them. A service really probably shouldn't be a dataclass